Source/WebCore/rendering/RenderLayerModelObject.cpp

```diff
@@ -291,6 +291,35 @@ void RenderLayerModelObject::updateLayerTransform()
         layer()->updateTransform();
 }
 
+bool RenderLayerModelObject::setReferenceBoxForPathOperations()
+{
+    auto pathOperation = style().offsetPath();
+    if (!pathOperation)
+        return true;
+    if (is<BoxPathOperation>(pathOperation)) {
+        auto& boxPathOperation = downcast<BoxPathOperation>(*pathOperation);
+        auto pathReferenceBoxRect = snapRectToDevicePixelsIfNeeded(referenceBoxRect(boxPathOperation.referenceBox()), *this);
+        boxPathOperation.setPathForReferenceRect(FloatRoundedRect { pathReferenceBoxRect });
+    } else if (is<RayPathOperation>(pathOperation)) {
+        if (auto* parentBlock = containingBlock()) {
+            auto& rayPathOperation = downcast<RayPathOperation>(*pathOperation);
+            auto pathReferenceBoxRect = snapRectToDevicePixelsIfNeeded(parentBlock->transformReferenceBoxRect(parentBlock->style()), *this);
+            if (auto* containingBlockFlow = dynamicDowncast<RenderBlockFlow>(parentBlock)) {
+                if (parentBlock->needsLayout() && !containingBlockFlow->wasMarkedNeedingTransformUpdate(this)) {
```

Comment:
It's a bit gross that this sets some state on the `containingBlockFlow`. Why not just change the code to always update motion path things after the containing block has finished layout?

Reply:
Changed to set this state in RenderLayer::updateTransform and moved the rest to MotionPathHelper.

Reply:
Not sure if `RenderLayer::updateTransform()` is the right place to set this state. We can either set it here or we have to set in every `RenderObject` subclass where we call `updateLayerTransform()` in the individual `layout()` functions.

Reply:
Moved the marking of needing a transform update to `RenderLayerModelObject::updateLayerTransform`.